src/dataloader.ts

```diff
@@ -79,8 +68,13 @@ export abstract class Loader {
    * abort if we find a matching folder or reach the source root; for example,
    * if docs/data exists, we won’t look for a docs/data.zip.
    */
-  static find(sourceRoot: string, targetPath: string, {useStale = false} = {}): Loader | undefined {
-    const exact = this.findExact(sourceRoot, targetPath, {useStale});
+  static find(
+    sourceRoot: string,
+    targetPath: string,
+    interpreters: Config["interpreters"],
+    {useStale = false} = {}
```

Comment:
Why is `useStale` passed as a named option, but `interpreters` is a positional argument? Because it’s required, whereas `useStale` has a default? I feel like we need a better convention here, as we have a number of internal functions that have lots of arguments. I might experiment with some alternatives. Maybe it shouldn’t be a blocker, but I’m still hung up on the plumbing.

Reply:
I don't any argument in favor of this style. Passing it as {useStale, interpreters} would be fine too.